internal/configure/disks/dc_item.go

```diff
@@ -0,0 +1,84 @@
+/*
+ *  Copyright (c) 2023 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: CurveAdm
+ * Created Date: 2023-02-24
+ * Author: Lijin Xiong (lijin.xiong@zstack.io)
+ */
+
+package disks
+
+import (
+	"github.com/opencurve/curveadm/internal/common"
+	comm "github.com/opencurve/curveadm/internal/configure/common"
+)
+
+const (
+	DEFAULT_FORMAT_PERCENT = 90
+)
+
+var (
+	itemset = comm.NewItemSet()
+
+	CONFIG_GLOBAL_CONTAINER_IMAGE = itemset.Insert(
+		common.DISK_FORMAT_CONTAINER_IMAGE,
+		comm.REQUIRE_STRING,
+		false,
+		nil,
+	)
+
+	CONFIG_GLOBAL_FORMAT_PERCENT = itemset.Insert(
+		common.DISK_FORMAT_PERCENT,
+		comm.REQUIRE_POSITIVE_INTEGER,
+		false,
+		DEFAULT_FORMAT_PERCENT,
+	)
+
+	CONFIG_GLOBAL_HOST = itemset.Insert(
+		common.DISK_HOST,
+		comm.REQUIRE_SLICE,
+		false,
+		nil,
+	)
+	CONFIG_DISK_DEVICE = itemset.Insert(
+		common.DISK_QUERY_DISK_DEVICE,
+		comm.REQUIRE_STRING,
+		false,
+		nil,
+	)
+
+	CONFIG_DISK_MOUNT_POINT = itemset.Insert(
+		common.DISK_FORMAT_MOUNT_POINT,
+		comm.REQUIRE_STRING,
+		false,
+		nil,
+	)
+
+	CONFIG_DISK_HOST_EXCLUDE = itemset.Insert(
+		common.DISK_HOST_EXCLUDE,
```

Comment:
Maybe we can remove the `host_exclude` and `host_only` configure item.

Reply:
What about the use case that disk number is different among hosts?

Reply:
Are there any good solutions to support different disk number among hosts?

Reply:
> What about the use case that disk number is different among hosts?

Hi, guys, for solving the problem that the disk list in each hosts is different while formatting, we have provide two designs which show as below:

Design 1: using `host_only` and `host_exclude`
---
```
global:
  format_percent: 95
  host:
    - curve-1
    - curve-2
    - curve-3
    
disk:
  - device: /dev/sdb1
    mount: /data/chunkserver0
  - device: /dev/sdc1
    mount: /data/chunkserver1
  - device: /dev/sdd1
    mount: /data/chunkserver2
    host_exclude:  # for the use case that all hosts have this disk device except some hosts
      - curve-3
  - device: /dev/sde1
    mount: /data/chunkserver3
    host_only:     # for the use case that only some hosts have this disk device
      - curve-1
      - curve-2
```
from @legionxiong

Design 2: using level design
---

```
global:
  format_percent: 95
  hosts:
    - curve-1
    - curve-2
    - curve-3
   
  disks:
    - device: /dev/sdb1
       mount: /data/chunkserver0
    - device: /dev/sdc1
       mount: /data/chunkserver1
       format_percent: 90
    - device: /dev/sdd1
       mount: /data/chunkserver2
       hosts:
         - curve-1
         - curve-2
    - device: /dev/sde1
       mount: /data/chunkserver3
       hosts:
         - curve1
```
pros:
* It might be more easier to understand.
* The [level design](https://github.com/opencurve/curveadm/wiki/topology#%E5%B1%82%E7%BA%A7) `SAME AS` our other configures, like topology.

what's your opinion? cc @aspirer @Cyber-SiKu @tsonglew

Reply:
Or different disk number among hosts is not expected for curvebs, but what about the different disk device path?

Reply:
`Design 2` is much easier than `host_only` use case 👍 , but for `host_exclude` use case it may be a bit painful. And disk devices among hosts may be changed from time to time.

Reply:
Combine `Design 1` and `Design 2`, we get `Design 3`:
```
global:
  format_percent: 95
  host:
    - curve-1
    - curve-2
    - curve-3
   
  disk:
    - device: /dev/sdb1
      mount: /data/chunkserver0
    - device: /dev/sdc1
      mount: /data/chunkserver1
      format_percent: 90
    - device: /dev/sdd1
      mount: /data/chunkserver2
      exclude:
        - curve-1
    - device: /dev/sde1
      mount: /data/chunkserver3
      host:
        - curve1
        - curve2
        
```

Reply:
I would prefer the 2nd and the 3rd among designs above. And it seems `hosts_exclude` is a more explicit name than `exclude`

Reply:
Done with `Desing 3`.

Reply:
> I would prefer the 2nd and the 3rd among designs above. And it seems `hosts_exclude` is a more explicit name than `exclude`

Agree, `hosts_exclude` is more explicit, but disk configure has nothing to `exclude` except `hosts`. @Wine93 How do you think about this?

Reply:
> > I would prefer the 2nd and the 3rd among designs above. And it seems `hosts_exclude` is a more explicit name than `exclude`
> 
> Agree, `hosts_exclude` is more explicit, but disk configure has nothing to `exclude` except `hosts`. @Wine93 How do you think about this?

As you said, it is still easy to understand if we clearly describe what exclude does in our WIKI, like [hosts](https://github.com/opencurve/curveadm/wiki/hosts#%E4%B8%BB%E6%9C%BA%E9%85%8D%E7%BD%AE%E9%A1%B9). I agree the design 3 shown above.

Reply:
OK, how to add / update curveadm WiKi? We need add a `disks` page.

Reply:
> OK, how to add / update curveadm WiKi? We need add a `disks` page.

You can refer to [how-can-i-make-a-pull-request-for-a-wiki-page-on-github](https://stackoverflow.com/questions/10642928/how-can-i-make-a-pull-request-for-a-wiki-page-on-github/11481887#11481887), and our wiki git address is https://github.com/opencurve/curveadm.wiki.git.

Reply:
> > OK, how to add / update curveadm WiKi? We need add a `disks` page.
> 
> You can refer to [how-can-i-make-a-pull-request-for-a-wiki-page-on-github](https://stackoverflow.com/questions/10642928/how-can-i-make-a-pull-request-for-a-wiki-page-on-github/11481887#11481887), and our wiki git address is https://github.com/opencurve/curveadm.wiki.git.

It seems I have no privilege to view curveadm wilki repo, got `404` :(